_profiles/drupal/resource-versioning/index.md

```diff
@@ -0,0 +1,301 @@
+
+Sophisticated servers may support multiple versions of a resource as well as
+multiple working copies of a resource (e.g., to support multiple
+languages or within a version control system which supports multiple branches).
+
+This profile creates a standard for JSON:API to support these diverse versioning
+schemes.
+# Query Parameter
+
+## Usage
+
+An endpoint **MAY** support a `resourceVersion` query parameter to allow a
+client to indicate which version(s) of a resource should be returned.
+
+If an endpoint does not support the `resourceVersion` parameter, it **MUST**
+respond with `400 Bad Request` to any requests that include it. An error object
+detailing the source of the error **SHOULD** include a `source` member referencing
+the `resourceVersion` query parameter.
+
+If an endpoint supports the `resourceVersion` parameter and a client supplies
+it:
+
+  - The server’s response **MUST** contain the most appropriate version of the
+    resource requested.
+  - The server **MUST** respond with `404 Not Found` if an appropriate version of
+    the resource requested cannot be located. 
+    
+> Note: This means that a server should not provide "fallbacks" unless the
+> behavior is well defined by the version negotiation mechanism (see below).
+> These rules apply to individual and collection endpoints alike.
+
+## Format
+  
+The value of the `resourceVersion` parameter **MUST** be a colon-separated
+(U+003A COLON, “:”) string composed of one or more segments. The first segment
+of the string **MUST** be interpreted as an identifier for a _version negotiation
+mechanism_. A version negotiation mechanism defines how a server will
+locate an appropriate resource version. Subsequent segments of the string
+**SHOULD** be interpreted as version negotiation arguments for the preceding
+mechanism. Collectively, this query parameter value is known as the _version
+identifier_.
+
+> Note: For example, a server may support both ID-based and time-based
+> mechanisms for requesting a resource version. The former mechanism would be
+> useful for comparing versions and the latter could be useful for requesting a
+> resource as it existed at an arbitrary point in time. This profile does not
+> attempt to define every possible mechanism for versioning resources.
+
+```
+                 version-identifier
+                  _______|________
+                 /                \
+?resourceVersion=rel:latest-version
+                 \_/ \____________/
+                  |        |
+        version-negotiator |
+                    version-argument
+```
+
+## Server Responsibilities
+
+<a id="bad-version-negotiator"></a>A server **MUST** respond with `400 Bad Request` if a version negotiator is not
+supported. An error object detailing the source of the error **SHOULD** include a
+`type` link to
+`https://jsonapi.org/profiles/drupal/resource-versioning/#bad-version-negotiator`
+and **SHOULD** include a `source` member referencing the `resourceVersion` query
```

Comment:
I think at least one of these two `**SHOULD**`s should be come a `**MUST**`. Otherwise a `400 Bad Request` response without any useful information is allowed, which is bad DX.